javascript/index.js

```diff
@@ -4,37 +4,127 @@
 // 🚨🚨🚨 Comment out the below code before you start working on the code
 
 // Out of sync
-  getInstruction("mashedPotatoes", 0, (step1) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
-  }, (error) => console.log(error));
+  // getInstruction("mashedPotatoes", 0, (step1) => {
+  //   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+  // }, (error) => console.log(error));
   
-  getInstruction("mashedPotatoes", 1, (step2) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
-  }, (error) => console.log(error));
+  // getInstruction("mashedPotatoes", 1, (step2) => {
+  //   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+  // }, (error) => console.log(error));
   
-  getInstruction("mashedPotatoes", 2, (step3) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
-  }, (error) => console.log(error));
+  // getInstruction("mashedPotatoes", 2, (step3) => {
+  //   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+  // }, (error) => console.log(error));
   
-  getInstruction("mashedPotatoes", 3, (step4) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
-  }, (error) => console.log(error));
+  // getInstruction("mashedPotatoes", 3, (step4) => {
+  //   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+  // }, (error) => console.log(error));
   
-  getInstruction("mashedPotatoes", 4, (step5) => {
-    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
-    document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
+  // getInstruction("mashedPotatoes", 4, (step5) => {
+  //   document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+  //   document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+  // }, (error) => console.log(error));
 
 
 
 // Iteration 1 - using callbacks
 // ...
+function addListElement(Selector,val){
+  document.querySelector(Selector).innerHTML += `<li>${val}</li>`
+}
+
+function selectorUnhide(Selector){
+  document.querySelector(Selector).removeAttribute("hidden");
+}
+
+function func1(step,callback) {
+  const htmlEdit = (val) => addListElement("#mashedPotatoes",val);
+
+  getInstruction( "mashedPotatoes", step,
+    (val) => {
+      htmlEdit(val);
+      callback();
+    },
+    (error) => console.log(error)
+  );
+}
+
+func1(0, () => {
```

Comment:
you could think about how this could be a recursive function (a function that calls itself) and increment itself 1 by 1 in the callback. instead of calling callback within the callback passed to getInstruction you could instead call func1(step+1) 
also, if you do this for practice - make sure to include a if > maxSteps - return so that you don't recurse infinitely